src/blueteam/LumberJack.java

```diff
@@ -32,54 +33,127 @@
-			// Use strike() to hit all nearby robots!
-			rc.strike();
+		RobotInfo[] robots = rc.senseNearbyRobots(GameConstants.LUMBERJACK_STRIKE_RADIUS, enemy);
+		// See if there are any enemy robots within striking range
+		strikeNearRobot(robots);
+
+		TreeInfo[] trees = rc.senseNearbyTrees();
+
+		for (TreeInfo treeInfo : trees) {
+			if (rc.canChop(treeInfo.ID) && treeInfo.team != rc.getTeam()) {
+				rc.chop(treeInfo.ID);
+				return;
+			}
+		}
+
+		// trying to find some trees with robot inside. The priority is to cut
+		// down this tree.
+		Optional<TreeInfo> bonusTree = Arrays.stream(trees).filter(tree -> tree.getContainedRobot() != null)
+				.findFirst();
+		if (bonusTree.isPresent()) {
+			moveDirection = myLocation.directionTo(bonusTree.get().getLocation());
+			rc.setIndicatorDot(myLocation, 255, 0, 0);
+			if (rc.canMove(moveDirection)) {
+				tryMove(moveDirection);
+			} else {
+				// two robots are behind each other and both are trying to
+				// access the same tree. This let's them avoid each other
+				if (rand.nextFloat() < 0.5)
+					tryMove(moveDirection.rotateLeftDegrees(90));
+				else
+					tryMove(moveDirection.rotateRightDegrees(90));
+			}
 			return;
+
 		}
+
 		// No close robots, so search for robots within sight radius
-		robots = rc.senseNearbyRobots(-1, enemy);
+		Optional<RobotInfo> nearRobot = Arrays.stream(rc.senseNearbyRobots(-1, enemy))
+				.filter(robot -> robot.getType() != RobotType.SOLDIER && robot.getType() != RobotType.TANK).findFirst();
 		// If there is a robot, move towards it
-		if (robots.length > 0) {
+		if (nearRobot.isPresent()) {
 
-			MapLocation enemyLocation = robots[0].getLocation();
+			MapLocation enemyLocation = nearRobot.get().getLocation();
 			Direction toEnemy = myLocation.directionTo(enemyLocation);
 			if (tryMove(toEnemy))
 				return;
 		}
-		TreeInfo[] trees = rc.senseNearbyTrees();
-		for (TreeInfo treeInfo : trees) {
-			if (rc.canChop(treeInfo.ID) && treeInfo.team != rc.getTeam()) {
-				rc.chop(treeInfo.ID);
+
+		// no tree can be chopped -> go to the nearest (enemy) tree:
+		for (TreeInfo tree : trees) {
+			if (tree.getTeam() == rc.getTeam())
+				continue;
+			Direction dirToTree = myLocation.directionTo(tree.getLocation());
+			if (rc.canMove(dirToTree)) {
+				tryMove(dirToTree);
+				rc.setIndicatorDot(tree.getLocation(), 0, 255, 0);
 				return;
 			}
 		}
-		// no tree can be chopped -> go to the nearest (enemy) tree:
-		Optional<TreeInfo> nearest = Arrays.stream(trees).filter(x -> x.team != rc.getTeam()).findFirst();
-		if (nearest.isPresent()) {
-			Direction dirToTree = myLocation.directionTo(nearest.get().getLocation());
-			tryMove(dirToTree);
-			rc.setIndicatorDot(myLocation, 0, 255, 0);
-			return;
-		}
 		if (rc.hasMoved())
 			return;
-		// No tree in sight -> move randomly:
+		// No tree in sight -> move randomly or attack:
+		MapLocation oponentInitially = rc.getInitialArchonLocations(rc.getTeam().opponent())[0];
```

Comment:
I don't want to go to random archon. I want to go to the closest one. It takes long time for lumberjack to go to archon location because he is clearing all the trees on the way there. I expect that array which is returned by getInitialArchonLocations() is sorted by distance from my location. This is usually true in all methods in this framework.